protos/flyteidl/admin/signal.proto

```diff
@@ -0,0 +1,87 @@
+syntax = "proto3";
+
+package flyteidl.admin;
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin";
+
+import "flyteidl/admin/common.proto";
+import "flyteidl/core/identifier.proto";
+import "flyteidl/core/literals.proto";
+import "flyteidl/core/types.proto";
+
+// SignalGetOrCreateRequest represents a request structure to retrive or create a signal.
+// See :ref:`ref_flyteidl.admin.Signal` for more details
+message SignalGetOrCreateRequest {
+    // A unique identifier for the requested signal.
+    core.SignalIdentifier id = 1;
+
+    // A type denoting the required value type for this signal.
+    core.LiteralType type = 2;
+}
+
+// SignalListRequest represents a request structure to retrieve a collection of signals.
+// See :ref:`ref_flyteidl.admin.Signal` for more details
+message SignalListRequest {
+    // Indicates the workflow execution to filter by.
+    // +required
+    core.WorkflowExecutionIdentifier workflow_execution_id = 1;
+
+    // Indicates the number of resources to be returned.
+    // +required
+    uint32 limit = 2;
+
+    // In the case of multiple pages of results, the, server-provided token can be used to fetch the next page
+    // in a query.
+    // +optional
+    string token = 3;
+
+    // Indicates a list of filters passed as string.
+    // More info on constructing filters : <Link>
```

Comment:
nit: can we populate the `<Link>` with a sphinx reference

Reply:
To be honest, I just copied the list proto from other messages. It doesn't look like any have a link populated. Does one exist? Otherwise I will just remove the link.